plugins/wasm-go/extensions/ai-cache/core.go

```diff
@@ -0,0 +1,261 @@
+		handleQueryResults(key, results, ctx, log, stream, c, err)
+	})
+}
+
+// performEmbeddingQuery executes the embedding-based similarity search.
+func performEmbeddingQuery(key string, ctx wrapper.HttpContext, c config.PluginConfig, log wrapper.Log, stream bool) error {
+	embeddingQuerier, ok := c.GetVectorProvider().(vector.EmbeddingQuerier)
+	if !ok {
+		return logAndReturnError(log, fmt.Sprintf("[performEmbeddingQuery] active vector provider does not implement EmbeddingQuerier interface"))
+	}
+
+	activeEmbeddingProvider := c.GetEmbeddingProvider()
+	if activeEmbeddingProvider == nil {
+		return logAndReturnError(log, fmt.Sprintf("[performEmbeddingQuery] no embedding provider configured for similarity search"))
+	}
+
+	return activeEmbeddingProvider.GetEmbedding(key, ctx, log, func(textEmbedding []float64, err error) {
+		log.Debugf("[%s] [performEmbeddingQuery] GetEmbedding success, length of embedding: %d, error: %v", PLUGIN_NAME, len(textEmbedding), err)
+		if err != nil {
+			handleInternalError(err, fmt.Sprintf("[%s] [performEmbeddingQuery] error getting embedding for key: %s", PLUGIN_NAME, key), log)
+			return
+		}
+		ctx.SetContext(CACHE_KEY_EMBEDDING_KEY, textEmbedding)
+
+		err = embeddingQuerier.QueryEmbedding(textEmbedding, ctx, log, func(results []vector.QueryResult, ctx wrapper.HttpContext, log wrapper.Log, err error) {
+			handleQueryResults(key, results, ctx, log, stream, c, err)
+		})
+		if err != nil {
+			handleInternalError(err, fmt.Sprintf("[%s] [performEmbeddingQuery] error querying vector database for key: %s", PLUGIN_NAME, key), log)
+		}
+	})
+}
+
+// handleQueryResults processes the results of similarity search and determines next actions.
+func handleQueryResults(key string, results []vector.QueryResult, ctx wrapper.HttpContext, log wrapper.Log, stream bool, c config.PluginConfig, err error) {
+	if err != nil {
+		handleInternalError(err, fmt.Sprintf("[%s] [handleQueryResults] error querying vector database for key: %s", PLUGIN_NAME, key), log)
+		return
+	}
+
+	if len(results) == 0 {
+		log.Warnf("[%s] [handleQueryResults] no similar keys found for key: %s", PLUGIN_NAME, key)
+		proxywasm.ResumeHttpRequest()
+		return
+	}
+
+	mostSimilarData := results[0]
+	log.Debugf("[%s] [handleQueryResults] for key: %s, the most similar key found: %s with score: %f", PLUGIN_NAME, key, mostSimilarData.Text, mostSimilarData.Score)
+	simThreshold := c.GetVectorProviderConfig().Threshold
+	simThresholdRelation := c.GetVectorProviderConfig().ThresholdRelation
+	if compare(simThresholdRelation, mostSimilarData.Score, simThreshold) {
+		log.Infof("[%s] key accepted: %s with score: %f", PLUGIN_NAME, mostSimilarData.Text, mostSimilarData.Score)
+		if mostSimilarData.Answer != "" {
+			// direct return the answer if available
+			processCacheHit(key, mostSimilarData.Answer, stream, ctx, c, log)
```

Comment:
好的，现在的逻辑修改为：如果在向量存储的最相似结果中含有Answer字段则使用并保存在缓存中，若不含Answer字段且配置了cache服务，则继续在缓存中查找是否有Answer的缓存
```go
		log.Infof("[%s] key accepted: %s with score: %f", PLUGIN_NAME, mostSimilarData.Text, mostSimilarData.Score)
		if mostSimilarData.Answer != "" {
			// direct return the answer if available
			processCacheHit(key, mostSimilarData.Answer, stream, ctx, c, log)
			cacheResponse(ctx, c, key, mostSimilarData.Answer, log)
		} else {
			if c.GetCacheProvider() != nil {
				CheckCacheForKey(mostSimilarData.Text, ctx, c, log, stream, false)
			} else {
				// Otherwise, do not check the cache, directly return
				log.Infof("[%s] cache hit for key: %s, but no corresponding answer found in the vector database", PLUGIN_NAME, mostSimilarData.Text)
				proxywasm.ResumeHttpRequest()
			}
		}```